client/components/app-promo/lib/promo-retriever.js

```diff
@@ -0,0 +1,34 @@
+import { sample } from 'lodash';
+
+const promo_options = [
+	{
+		promoCode: 'a0001',
+		message: 'WordPress.com your way — desktop app now available for Mac, Windows, and Linux.'
+	},
+	{
+		promoCode: 'a0002',
+		message: 'Get WordPress.com app for your desktop.'
+	},
+	{
+		promoCode: 'a0003',
+		message: 'WordPress.com app now available for desktop.'
+	},
+	{
+		promoCode: 'a0005',
+		message: 'WordPress.com at your fingertips — download app for desktop.'
+	},
+	{
+		promoCode: 'a0006',
+		message: 'WordPress.com in the palm of your hands — download app for mobile.'
+	}
+];
+
+export const getRandomPromo = ( ) => sample ( promo_options );
+
+export const getPromoLink = ( location, promoCode ) => {
+	const promoType = 'a0006' === promoCode
```

Comment:
For the purposes of scaling this, the type may best live as a property of the option itself. Assigning the value of `promoOptions` as an object would simplify lookup. Better still, Lodash's `sample` will happily accept an object and return a random property value, so there should be no impact to `getRandomPromo`.

```js
const promoOptions = {
	// ...
	a0006: {
		promoCode: 'a0006',
		message: 'WordPress.com ...',
		type: 'mobile'
	}
};

export const getPromoLink = ( location, promoCode ) => {
	const { promoType } = promoOptions[ promoCode ];
	return `...`;
};
```